wrappers/rust/icicle-curves/icicle-bw6-761/src/curve.rs

```diff
@@ -0,0 +1,22 @@
+#[cfg(feature = "arkworks")]
+use ark_bw6_761::{g1::Config as ArkG1Config, Fq, Fr};
+use icicle_core::curve::{Affine, CurveConfig, Projective};
+use icicle_core::field::{Field, FieldConfig};
+use std::ffi::{c_uint, c_void};
+
+impl_curve!(12, 24,);
```

Comment:
For BW6-761 curve it might be better to import BLS12-377 base field instead of default macro implementation of the scalar field (see arkworks [version](https://github.com/arkworks-rs/algebra/blob/master/curves/bw6_761/src/fields/fr.rs)). I imagine that any codebase that uses this curve will benefit from compile-time knowledge that the two fields are the same. I think this also warrants splitting `imps_curve!` macro into two: `impl_field!` and `impl_curve!` which I think is a good idea on its own: a macro for implementing stand-alone fields will be useful for STARK fields. Also, consider how G2 works in BW6-761: it's a curve over the same base field as G1 (but a different curve nonetheless). I think to accommodate this it's nice to have a separate macro for fields. Also maybe one idea for macros would be not relying on names available in the scope but passing them as parameters (though I guess it's a tradeoff and you can argue that having concrete names inside a macro is more readable).

Reply:
Separated field macros from curve macros in 98675f766e432b373cf024a04cf5bb6fd32befa4

Reply:
Implementing bw6-761 with bls12-377 base field seems impossible in the current design due to this line in NTT macro
`impl NTT<$field> for $field_config {`

The base field config for bls12-377 is defined in another crate, so It's impossible to define new trait implementation here

Reply:
> The base field config for bls12-377 is defined in another crate, so It's impossible to define new trait implementation here

Yes sure, we should implement NTT on the base field of bls12-377 in bls12-377 crate IMO

Reply:
Oh wait, I don't think the second option works. After all, bls12-377 create will be compiled first so linker won't be able to find the extern NTT function which only gets created before bw6-761 crate is built...

Reply:
Moved this to an issue #307